### PR TITLE
IGNITE-6564 Incorrect calculation size and keySize for cluster cache metrics

### PR DESCRIPTION
**Pull request by a-polyakov**
Signed-off-by: a-polyakov <polyakov.alexandr.alexandrovich@gmail.com>

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/CacheMetricsImpl.java [hunk 1]
```diff
@@ -1042,9 +1034,7 @@ public EntriesStatMetrics getEntriesStat() {
         stat.offHeapPrimaryEntriesCount(offHeapPrimaryEntriesCnt);
         stat.offHeapBackupEntriesCount(offHeapBackupEntriesCnt);
         stat.heapEntriesCount(heapEntriesCnt);
-        stat.size(size);
```
Comment (alamar) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/CacheMetricsImpl.java]: Why do we do that?

File: modules/core/src/main/java/org/apache/ignite/cache/CacheMetrics.java [hunk 2]
```diff
@@ -315,6 +315,7 @@
 
     /**
      * Gets number of non-{@code null} values in the cache.
+     * Note this method will always return {@code 0}
```
Comment (alamar) [inline modules/core/src/main/java/org/apache/ignite/cache/CacheMetrics.java]: Why? Is this an incompatible change?

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/CacheMetricsSnapshot.java [hunk 3]
```diff
@@ -357,9 +353,7 @@ public CacheMetricsSnapshot(CacheMetricsImpl m) {
 
         offHeapAllocatedSize = m.getOffHeapAllocatedSize();
 
-        size = entriesStat.size();
```
Comment (alamar) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/CacheMetricsSnapshot.java]: Why are we changing this class at all if we introduce CacheMetricsSnapshotV2?

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/CacheMetricsSnapshot.java [hunk 4]
```diff
@@ -27,7 +27,9 @@
 
 /**
  * Metrics snapshot.
+ * @deprecated Replaced by CacheMetricsSnapshotV2 with versioning support.
  */
+@Deprecated
 public class CacheMetricsSnapshot implements CacheMetrics, Externalizable {
```
Comment (alamar) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/CacheMetricsSnapshot.java]: Can we also rollback entryProcessor* externationalization from this class so that https://issues.apache.org/jira/browse/IGNITE-10925 will be fixed even for  CacheMetricsSnapshot (v1) users?

File: modules/core/src/test/java/org/apache/ignite/testframework/junits/common/GridCommonAbstractTest.java [hunk 5]
```diff
@@ -2041,4 +2043,36 @@ protected void checkFutures() {
                 fail("Some transaction are not finished");
         }
     }
+
+    /**
+     * Wait for {@link EventType#EVT_NODE_METRICS_UPDATED} event will be receieved.
+     *
+     * @param countUpdate Number of events.
+     */
+    protected void awaitMetricsUpdate(int countUpdate) throws InterruptedException {
```
Comment (alamar) [inline modules/core/src/test/java/org/apache/ignite/testframework/junits/common/GridCommonAbstractTest.java]: Can you please split awaitMetricsUpdate refactoring in a separate PR?

**Comment by dspavlov** [comment]
The ticket is resolved, closing PR. 
 
Please close PRs for your resolved tickets and other non-needed PRs. 
 
Alternatively, ask a committer 
- to use GitHub UI (committer can associate accounts here: https://gitbox.apache.org/setup/ )
- or /scripts/apply_pull_request.sh from Apache Ignite source.
- or mention in commit message 'Fixes #(a number of PR)'. 
 
These methods will automatically close PR.
 
Feel free to reopen this PR if it is actual.
